core/domain/learner_progress_services.py

```diff
@@ -616,3 +650,114 @@ def get_collection_summary_dicts(collection_summaries):
         })
 
     return summary_dicts
+
+
+def get_activity_progress(user_id):
+    """Returns the progress of the learners - the explorations and collections
+    completed by the user and those in progress.
+
+    Args:
+        user_id: str. The id of the learner.
+
+    Returns:
+        LearnerProgress. The learner progress domain object corresponding to the
+            particular learner.
+        dict. The number of entities that are no longer present.
+        list(str). The titles of the collections to which new explorations have
+            been added.
+
+    """
+    incomplete_exploration_ids = get_all_incomplete_exp_ids(user_id)
+    incomplete_collection_ids = get_all_incomplete_collection_ids(user_id)
+    completed_collection_ids = get_all_completed_collection_ids(user_id)
+    completed_exploration_ids = get_all_completed_exp_ids(user_id)
+
+    incomplete_exploration_keys = (
+        datastore_services.get_keys_from_ids_and_model(
+            incomplete_exploration_ids, 'ExpSummaryModel'))
+    incomplete_collection_keys = (
+        datastore_services.get_keys_from_ids_and_model(
+            incomplete_collection_ids, 'CollectionSummaryModel'))
+    completed_exploration_keys = (
+        datastore_services.get_keys_from_ids_and_model(
+            completed_exploration_ids, 'ExpSummaryModel'))
+    completed_collection_keys = (
+        datastore_services.get_keys_from_ids_and_model(
+            completed_collection_ids, 'CollectionSummaryModel'))
+
+    activity_models = datastore_services.fetch_multiple_entities_by_keys(
```

Comment:
Maybe better: have just one function that takes an arg that looks something like this:

```
[
    ['ExpSummaryModel', incomplete_exploration_ids],
    ['CollectionSummaryModel', incomplete_collection_ids],
    ...
]
```

and have it return a list of lists of entities. I.e. do the list processing in the platform layer, since it's more related to the details of the datastore GET than it is to the learner dashboard.